### PR TITLE
Add integration test for sending from swap

### PR DESCRIPTION
**Pull request by QureshiFaisal**
This is in reference to issue #93 
1. Refer to the `send_from_swap.rs` in `tests`.
2. Kindly verify the logic for swapping coins between a taker & maker, asserting balances, listing UTXOs from the taker's wallet, creating a transaction and signing it.
3. I am unable to figure out the logic for broadcasting the signed transaction.

### CODE REVIEW

**Comment by rajarshimaitra** [comment]
Concept ACK.. reviewing soon..

**Review by rajarshimaitra — CHANGES_REQUESTED** [review]
Good attempt. But needs more work. 

Remove the usual checks already covered in standard swap. Filter one swapcoin from the list unspent and list and try to spend only that input. 

Broadcast the tx after signing. 

Confirm the tx. Then check that the new tx also appeared in the listunpsent output. Assert the wallet final balance with expected balance.

File: tests/send_from_swap.rs [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+ 
+    info!("All checks successful. Terminating integration test case");
+
+    //Step 1: Get swapcoin utxos in the taker wallet
+let swapcoin_utxos = taker
+    .read()
+    .unwrap()
+    .get_wallet()
+    .list_unspent_from_wallet(false, false)
+    .unwrap();
```
[the remark this answers is not in the document]
Comment (QureshiFaisal) [reply tests/send_from_swap.rs:195]: Here I collect all the unspent utxos, right below this I choose the txid and vout of the first transaction i.e `utxo_outpoint` to use only that specific transaction. So, I'm passing only one utxo from the list to `create_direct_send`

File: tests/send_from_swap.rs [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+    let utxo_outpoint = CoinToSpend::LongForm(OutPoint {
+        txid: utxo_entry.txid,
+        vout: utxo_entry.vout,
+    });
+
+// Extract UTXOSpendInfo from the unspent inputs
+let unspent_inputs: Vec<UTXOSpendInfo> = swapcoin_utxos
+    .iter()
+    .map(|(entry, spend_info)| spend_info.clone())
+    .collect();
```
[the remark this answers is not in the document]
Comment (QureshiFaisal) [reply tests/send_from_swap.rs:125]: `sign_transaction` needs a vector that iterates over `UTXOspendinfo` and thus I have created that vector in `unspent_inputs`

File: tests/send_from_swap.rs [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+let auth = Auth::None;
+
+
+let client = Client::new(rpc_url, auth).expect("Failed to create RPC client");
+
+// Use the sendrawtransaction RPC method to broadcast the transaction
+let txid = client.send_raw_transaction(raw_tx_hex).expect("Failed to broadcast transaction");
+
+println!("Transaction broadcasted successfully. Transaction ID: {}", txid);
+
```
Comment (QureshiFaisal) [inline tests/send_from_swap.rs:170]: This is the new broadcast logic

File: tests/send_from_swap.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (rajarshimaitra) [reply tests/send_from_swap.rs:125]: If you just pass in `[spend_info]` that will work.

File: tests/send_from_swap.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (rajarshimaitra) [reply tests/send_from_swap.rs:111]: How do you ensure that the first coin is always a swapcoin? Could be a seedcoin or fidelitycoin too. Later on also we are not checking if we are spending a swapcoin or some other kind of coin.

That needs to be checked here. And the way to do that is use `find` on the list_unspent result where spend_info == SwapCoin. 

You can use `matches!()` macro inside `find` to match the `spend_info` variable with a `SwapCoin` structure.

File: tests/send_from_swap.rs [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+let rpc_url = "http://localhost/".into();
+let auth = Auth::None;
+
+
+let client = Client::new(rpc_url, auth).expect("Failed to create RPC client");
+
+// Use the sendrawtransaction RPC method to broadcast the transaction
+let txid = client.send_raw_transaction(raw_tx_hex).expect("Failed to broadcast transaction");
+
+println!("Transaction broadcasted successfully. Transaction ID: {}", txid);
```
Comment (rajarshimaitra) [inline tests/send_from_swap.rs:169]: The rpc client is already available in the `test_framework` variable. Just do `test_framework.bitcoind.client`. You don't have to recreate a new client.

File: tests/send_from_swap.rs [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+        .unwrap()
+        .get_wallet()
+        .balance(false, false)
+        .unwrap();
+
+    
+    assert!(final_balance < Amount::from_btc(0.15).unwrap());
+} else {
+    panic!("No swapcoin utxos found in the taker wallet");
+}
```
Comment (rajarshimaitra) [inline tests/send_from_swap.rs:185]: let's assert the exact balance here, not the below-ness.

**Review by rajarshimaitra — CHANGES_REQUESTED** [review]
Few more comments. 

Also, run `cargo fmt` to get the formatting sorted.

**Pull request closed without merging**